backend/src/routes/v0/thread.py

```
@@ -357,3 +364,167 @@ async def delete_thread(
+                )
+
+            # Validate each decision against the interrupt's allowed_actions
+            for i, decision in enumerate(request.decisions):
+                if i < len(interrupts):
+                    interrupt = interrupts[i]
+                    if decision.decision_type not in interrupt.config.allowed_actions:
+                        allowed = [a.value for a in interrupt.config.allowed_actions]
+                        raise HTTPException(
+                            status_code=status.HTTP_400_BAD_REQUEST,
+                            detail=f"Decision type '{decision.decision_type.value}' not allowed. Allowed actions: {allowed}",
+                        )
```

Ruff (0.14.13)

Abstract `raise` to an inner function

(TRY301)